frontends/web/src/components/forms/button.test.tsx

```diff
@@ -0,0 +1,24 @@
+import { MemoryRouter } from 'react-router-dom';
+import { render, screen } from '@testing-library/react';
+import { ButtonLink, Button } from './button';
+
+describe('components/forms/button', () => {
+  describe('ButtonLink', () => {
+    it('renders as button when disabled with disabled attribute', () => {
+      render(<ButtonLink to="/settings" disabled>A ButtonLink</ButtonLink>);
+      expect(screen.getByRole('button', { name: /A ButtonLink/i })).toHaveAttribute('disabled');
+    });
+
+    it('renders as link when not disabled', () => {
+      render(<ButtonLink transparent to={'/settings'}>A ButtonLink</ButtonLink>, { wrapper: MemoryRouter });
+      expect(screen.getByRole('link', { name: /A ButtonLink/i })).toBeInTheDocument();
```

Comment:
I tried to find the documentation for toBeInTheDocument, found this https://testing-library.com/docs/guide-disappearance/ but it seems to be about async rendering/wait-utility. All I found about MemoryRouter is this example, https://testing-library.com/docs/example-react-router/ is this needed?

how does this work and is there a better link to the documentation?

Reply:
The reason why we need a wrapper here (which is the `MemoryRouter`)  is because this will render a `<Link/>` component from `react-router-dom`. And when rendering only a component on itself like this (the nature of unit tests), it is missing the "routing context" (in the app we use `BrowserRouter`).

So, we can also use `BrowserRouter` here, but `MemoryRouter` is preferred because it keeps the routing history independent of any external source.

Good sources to read (small reads):
1. https://learnwithparam.com/blog/different-types-of-router-in-react-router/ (what is MemoryRouter in simple words)
2. https://testing-library.com/docs/react-testing-library/api/#wrapper (what is a wrapper) -> supported by RTL since it's such a common usage
3. https://stackoverflow.com/questions/70904665/difference-between-memoryrouter-and-router-concerning-testing-with-react-testing (BrowserRouter vs MemoryRouter in RTL)
4. https://stackoverflow.com/questions/70805929/how-to-fix-error-usehref-may-be-used-only-in-the-context-of-a-router-compon